### PR TITLE
CSP-19 Implements provider course edit contact details

### PR DESCRIPTION
**Pull request by DevenKShah**

### CODE REVIEW

**Comment by DevenKShah** [comment]
I have introduced an asp.net middleware for validation instead of the mediator pipeline. This allows us to work with view model better. So instead of creating a validator on query/command like in inner api. you can use view model. The validator and view model both live in the web project. 
Also I have added partial view for validation summary that will work with the middleware and give you the links that will allow navigating to failing field.  I have added tag helpers that will allow to display

File: src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Controllers/EditCourseContactDetailsControllerTests/EditCourseContactDetailsControllerGetTests.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        {
+            _mediatorMock
+                .Setup(m => m.Send(It.IsAny<GetStandardDetailsQuery>(), It.IsAny<CancellationToken>()))
+                .ReturnsAsync((GetStandardDetailsQueryResult)null);
+
+            Func<Task> action = () => _sut.Index(larsCode);
+
+            await action.Should().ThrowAsync<InvalidOperationException>();
+        }
+
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Controllers/EditCourseContactDetailsControllerTests/EditCourseContactDetailsControllerGetTests.cs:86]: remove spaces

File: src/SFA.DAS.Roatp.CourseManagement.Application/Standards/Commands/UpdateContactDetails/UpdateProviderCourseContactDetailsCommand.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    public class UpdateProviderCourseContactDetailsCommand : IRequest
+    {
+        public int Ukprn{ get; set; }
+        public int LarsCode { get; set; }
+        public string UserId { get; set; }
+        public string ContactUsEmail { get; set; }
+        public string ContactUsPhoneNumber { get; set; }
+        public string ContactUsPageUrl { get; set; }
+        public string StandardInfoUrl { get; set; }
+
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.CourseManagement.Application/Standards/Commands/UpdateContactDetails/UpdateProviderCourseContactDetailsCommand.cs:14]: remove space

File: src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Validators/EditCourseContactDetailsViewModelValidatorTests/ValidateContactUsPageUrlTEsts.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
+﻿using FluentValidation.TestHelper;
```
Comment (MarkFCain) [inline src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Validators/EditCourseContactDetailsViewModelValidatorTests/ValidateContactUsPageUrlTEsts.cs:1]: typo in filename 'TEsts' > 'Tests'

**Review by MarkFCain — APPROVED** [review]
some coverage needs adding and general snagging work around space removal and similar.  Nothing that will require further approval

File: src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Validators/EditCourseContactDetailsViewModelValidatorTests/ValidateContactUsPageUrlTEsts.cs (hunk 3 above)
Comment (DevenKShah) [reply src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Validators/EditCourseContactDetailsViewModelValidatorTests/ValidateContactUsPageUrlTEsts.cs:1]: Fixed

File: src/SFA.DAS.Roatp.CourseManagement.Application/Standards/Commands/UpdateContactDetails/UpdateProviderCourseContactDetailsCommand.cs (hunk 2 above)
Comment (DevenKShah) [reply src/SFA.DAS.Roatp.CourseManagement.Application/Standards/Commands/UpdateContactDetails/UpdateProviderCourseContactDetailsCommand.cs:14]: done

File: src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Validators/EditCourseContactDetailsViewModelValidatorTests/ValidatePhoneNumberTests.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+            var command = new EditCourseContactDetailsViewModel()
+            {
+                ContactUsPhoneNumber = phoneNumber
+            };
+
+            var result = sut.TestValidate(command);
+
+            result.ShouldHaveValidationErrorFor(c => c.ContactUsPhoneNumber).WithErrorMessage(EditCourseContactDetailsViewModelValidator.TelephoneLengthErrorMessage);
+        }
```
[the remark this answers is not in the document]
Comment (DevenKShah) [reply src/SFA.DAS.Roatp.CourseManagement.Web.UnitTests/Validators/EditCourseContactDetailsViewModelValidatorTests/ValidatePhoneNumberTests.cs:60]: Done

**Review by MarkFCain — APPROVED** [review]

**Review by MarkFCain — DISMISSED** [review]

**Pull request merged**